contracts/utils/cryptography/RSA.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {Math} from "../math/Math.sol";
+
+/**
+ *  TODO:
+ *  - Further optimize ?
+ *  - Write documentation
+ *
+ *  Inspired by Adrià Massanet's work: https://github.com/adria0/SolRsaVerify
+ */
+library RSA {
+    /**
+     * @dev Verifies a PKCSv1.5 SHA256 signature
+     * @param data to verify
+     * @param sig is the signature
+     * @param exp is the exponent
+     * @param mod is the modulus
+     */
+    function pkcs1Sha256(
+        bytes memory data,
+        bytes memory sig,
+        bytes memory exp,
+        bytes memory mod
+    ) internal view returns (bool) {
+        return pkcs1Sha256(sha256(data), sig, exp, mod);
+    }
+
+    /**
+     * @dev Verifies a PKCSv1.5 SHA256 signature
+     * @param digest is the sha256 of the data
+     * @param sig is the signature
+     * @param exp is the exponent
+     * @param mod is the modulus
+     */
+    function pkcs1Sha256(
+        bytes32 digest,
+        bytes memory sig,
+        bytes memory exp,
+        bytes memory mod
+    ) internal view returns (bool) {
+        unchecked {
+            // cache and check length
+            uint256 length = mod.length;
+            if (length < 0x40 || length != sig.length) {
```

Comment:
`length != sig.length` corresponds to the first step of [RFC 3447 in section 8.2.2](https://datatracker.ietf.org/doc/html/rfc8017#section-8.2.2)

> 1. Length checking: If the length of the signature S is not k octets, output "invalid signature" and stop.